src/visualizers/SeqColor.ts

```diff
@@ -0,0 +1,258 @@
+        this.subG.colorMode(this.sketch.HSB)
+        this.subG.noStroke()
+        this.subG.fill(0, 0, 100)
+        this.subG.rect(0, 0, 800, 90)
+        this.subG.fill('black')
+        let tmpX = 0
+        let tmpY = 0
+
+        //Show the color of every prime number
+        for (let i = 0; i < this.primeNum.length; i += 1) {
+            this.subG.fill('black')
+            this.subG.text(String(this.primeNum[i]), 10 + tmpX, 15 + tmpY)
+            this.subG.fill(colorMap.get(this.primeNum[i]), 100, 100)
+
+            this.subG.ellipse(35 + tmpX, 15 + tmpY, 10, 10)
+            tmpX += 50
+            if (tmpX >= 800) {
+                tmpX = 0
+                tmpY += 30
+            }
+        }
+
+        this.sketch.image(this.subG, 0, 700)
+
+        //this.sketch.noLoop()
+    }
+
+    undrawBox() {
+        if (this.subG) {
+            console.log('undraw')
+
+            //Draw a new black background to cover the description box
+            this.subG.fill('black')
+            this.subG.rect(0, 0, 800, 90)
+            this.sketch.image(this.subG, 0, 700)
+        }
+    }
+
+    //change the brightness of the circle
+    changeColor(difference: number, now: number) {
+        if ((now + difference) % 100 < now) {
+            return 100
+        } else {
+            return (now + difference) % 100
+        }
+    }
+
+    isPrime(n: number) {
+        for (let i = 2; i <= Math.sqrt(n); i++) {
+            if (n % i === 0) {
+                return false
+            }
+        }
+        return n > 1
+    }
+
+    //return a number which represents the color
+    primeFactors(n: number, total_prime: number) {
+        const factors = []
+        let factor_check = 2
+        //assign color to each prime number
+        const colorNum = 360 / Number(total_prime)
+
+        colorMap.set(1, 0)
+        let tmp = 0
+
+        while (n >= 2) {
+            if (n % factor_check == 0) {
+                factors.push(factor_check)
+                n = n / factor_check
+            } else {
+                factor_check++
+            }
+        }
+
+        for (let i = 0; i < this.primeNum.length; i++) {
+            if (colorMap.has(this.primeNum[i]) == false) {
+                tmp += colorNum
+                colorMap.set(this.primeNum[i], tmp)
+            }
+        }
+
+        //Combine color for each prime factors
+        let colorAll = -1
+        for (let i = 0; i < factors.length; i++) {
+            for (let j = 0; j < this.primeNum.length; j++) {
+                if (factors[i] == this.primeNum[j]) {
+                    if (colorAll == -1) {
+                        colorAll = colorMap.get(factors[i])
+                    } else {
+                        colorAll = (colorAll + colorMap.get(factors[i])) / 2
+                    }
+                }
+            }
+        }
+
+        return colorAll
+    }
+
+    //Make sure every value is unique in an array
+    valueDoesNotRepeat(n: number, arr: number[]) {
```

Comment:
You could use [the JavaScript `includes()` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) for this.